### DISCUSSION TITLE
[Feature]: Support Vue SSG (prerendering)

### DISCUSSION THREAD

**Discussion by MuTsunTsai** [web-infra-dev/rsbuild#2917] [General]
### What problem does this feature solve?

Some of my projects are using Vue SSG (see [[1](https://vuejs.org/guide/extras/ways-of-using-vue#jamstack-ssg)] [[2](https://vuejs.org/guide/scaling-up/ssr#ssr-vs-ssg)]) to improve loading performance, and I'm hoping that Rsbuild vue plugin can add support to this feature so that I may migrate those projects to Rsbuild.

Currently for webpack there is [Prerenderer](https://github.com/Tofandel/prerenderer) that does similar things, but it does not support Rspack.

### What does the proposed API look like?

Ideally it will be a one-option setup:
```js
pluginVue({
  prerendering: true,
});
```

**Comment by chenjiahan** [comment]
The SSG feature should be provided by frameworks such as Rspress or Nuxt. 

`@rsbuild/plugin-vue` will focus on providing primary support for Vue, like SFC compilation.

**Comment by MuTsunTsai** [comment]
@chenjiahan You have a point, but my projects are not using those frameworks though. Perhaps forking @prerenderer/webpack-plugin to make it work on Rspack?

**Comment by chenjiahan** [comment]
Are you using `@prerenderer/webpack-plugin` at present?

**Comment by MuTsunTsai** [comment]
@chenjiahan Nope. Those old projects of mine are not even using webpack in fact (but mainly with Gulp + esbuild, if that matters). I noticed that @prerenderer/webpack-plugin doesn't work as I started a new project with Rsbuild experimenting the features I need (so that I may migrate with confidence once all features are supported).

**Comment by chenjiahan** [comment]
The `@prerenderer/webpack-plugin` depends on `html-webpack-plugin`, so it can not be used in Rsbuild (Rsbuild uses the forked html-rspack-plugin). And I'm not sure whether @prerenderer/webpack-plugin supports Vue SSG.

For this requirement, I believe that the Rsbuild ecosystem needs more SSG frameworks or plugins similar to Rspress. I look forward to the community building such solutions. 😄

**Comment by lichenabson** [comment]
Yeah, I've implemented the solution using the rsbuild plugin. It wasn't too challenging, and it didn't cause me much trouble.

**Reply by MuTsunTsai** [reply]
Could you show us how?

**Comment by MuTsunTsai** [comment]
Alright, finally I've also succeeded in implementing Vue SSG with Rsbuild. The implementation is pretty much the same idea as the method used by Rspress. I've made it into a package, should anyone be interested:
https://www.npmjs.com/package/@mutsuntsai/rsbuild-plugin-vue-ssg

**Reply by chenjiahan** [reply]
Cool 👍
